src/ideations/ideations.controller.ts

```diff
@@ -0,0 +1,106 @@
+import {
+    Controller,
+    Get,
+    Param,
+    Post,
+    Body,
+    Patch,
+    Delete,
+    ParseIntPipe
+} from "@nestjs/common";
+import { IdeationsService } from "./ideations.service";
+import { CreateIdeationDto } from "./dto/create-ideation.dto";
+import { UpdateIdeationDto } from "./dto/update-ideation.dto";
+import { ApiCreatedResponse, ApiTags } from "@nestjs/swagger";
+import { Ideation } from "./entities/ideation.entity";
+import { CreateIdeationVoteDto } from "./dto/create-ideation-vote.dto";
+
+@Controller()
+@ApiTags("ideations")
+export class IdeationsController {
+    constructor(private readonly ideationsService: IdeationsService) {}
+
+    //TODO updated userId to grab uuid/JWT once 
+    @Post("users/:userId/teams/:teamId/ideations")
+    @ApiCreatedResponse({type: Ideation})
+    createIdeation(
+        @Param("userId") userId: string,
+        @Param("teamId", ParseIntPipe) teamId: number,
+        @Body() createIdeationDto: CreateIdeationDto,
+    ) {
+        return this.ideationsService.createIdeation(
+            teamId, 
+            userId, 
+            createIdeationDto
+        );
+    };
+
+    @Get("/teams/:teamId/ideations")
+    @ApiCreatedResponse({type: Ideation})
+    async getIdeationsByVoyageTeam(
+        @Param("teamId", ParseIntPipe) teamId: number
+    ) {
+        const projectIdeas = await this.ideationsService.getIdeationsByVoyageTeam(teamId);
+        return projectIdeas.map((projectIdea)=>{
+            return {
+                ...projectIdea, 
+                voteCount: projectIdea.projectIdeaVotes.length
+            }
+        })
+    };
+
+    @Patch("users/:userId/ideations/:ideationId")
```

Comment:
userID shouldn't need be passed in the endpoint url, i think it would come out from the auth/jwt eventually, for now we could just put in the body

I also think only the person who submitted the idea is able to update it, which means we'll need a check to see if the voyageMemberId passed in here matches the one in the database before we do the update.

I also think there might be a confusion here, it's "userId (Int)" in the table which actually is the "VoyageMemberId". I see you use usedId(string) here which is not correct

We use "VoyageMemberId" here and everything linked to a voyage participant because a Chingu member (user) can be in multiple voyages